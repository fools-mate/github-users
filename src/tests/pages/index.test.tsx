import { render } from '@testing-library/react';

import Home from '../../pages/index';

describe('Index page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should insert SearchUser view', () => {
    const { getByText } = render(<Home />);

    expect(getByText('Search:')).not.toBeNull();
  });
});
