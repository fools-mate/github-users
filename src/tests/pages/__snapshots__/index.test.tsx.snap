// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="Main-sc-5abt04-0 xuMSH"
    >
      <img
        alt="Inspectocat"
        class="Card__Inspectocat-sc-1xtfonb-1 gHDjhQ"
        src="/inspectocat.png"
      />
      <section
        class="Card__Container-sc-1xtfonb-0 cxcchT"
      >
        <h1
          class="Title-sc-1g805ww-0 loXuFm"
        >
          GitHub Users
        </h1>
        <div
          class="SearchBar__Container-hn5jjm-0 cRtDkb"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M337.509 305.372h-17.501l-6.571-5.486c20.791-25.232 33.922-57.054 33.922-93.257C347.358 127.632 283.896 64 205.135 64 127.452 64 64 127.632 64 206.629s63.452 142.628 142.225 142.628c35.011 0 67.831-13.167 92.991-34.008l6.561 5.487v17.551L415.18 448 448 415.086 337.509 305.372zm-131.284 0c-54.702 0-98.463-43.887-98.463-98.743 0-54.858 43.761-98.742 98.463-98.742 54.7 0 98.462 43.884 98.462 98.742 0 54.856-43.762 98.743-98.462 98.743z"
            />
          </svg>
          <label
            for="search-field"
          >
            Search:
          </label>
          <input
            class="InputField-sc-1yhz711-0 iHkRXZ"
            id="search-field"
            name="Search"
            placeholder="Please enter a username"
          />
        </div>
        <div>
          <div
            class="SearchInfoMsg__Container-p7aji8-0 ecRaNv"
          />
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Index page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="Main-sc-5abt04-0 xuMSH"
    >
      <img
        alt="Inspectocat"
        class="Card__Inspectocat-sc-1xtfonb-1 gHDjhQ"
        src="/inspectocat.png"
      />
      <section
        class="Card__Container-sc-1xtfonb-0 cxcchT"
      >
        <h1
          class="Title-sc-1g805ww-0 loXuFm"
        >
          GitHub Users
        </h1>
        <div
          class="SearchBar__Container-hn5jjm-0 cRtDkb"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M337.509 305.372h-17.501l-6.571-5.486c20.791-25.232 33.922-57.054 33.922-93.257C347.358 127.632 283.896 64 205.135 64 127.452 64 64 127.632 64 206.629s63.452 142.628 142.225 142.628c35.011 0 67.831-13.167 92.991-34.008l6.561 5.487v17.551L415.18 448 448 415.086 337.509 305.372zm-131.284 0c-54.702 0-98.463-43.887-98.463-98.743 0-54.858 43.761-98.742 98.463-98.742 54.7 0 98.462 43.884 98.462 98.742 0 54.856-43.762 98.743-98.462 98.743z"
            />
          </svg>
          <label
            for="search-field"
          >
            Search:
          </label>
          <input
            class="InputField-sc-1yhz711-0 iHkRXZ"
            id="search-field"
            name="Search"
            placeholder="Please enter a username"
          />
        </div>
        <div>
          <div
            class="SearchInfoMsg__Container-p7aji8-0 ecRaNv"
          />
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
